packages/flutter_tools/lib/src/windows/visual_studio.dart

```diff
@@ -128,23 +146,34 @@ class VisualStudio {
   /// This key is under the 'catalog' entry.
   static const String _catalogDisplayVersionKey = 'productDisplayVersion';
 
+  /// The key for a pre-release version.
+  ///
+  /// This key is under the 'catalog' entry.
+  static const String _isPrereleaseKey = 'productMilestoneIsPreRelease';
+
   /// Returns the details dictionary for the newest version of Visual Studio
   /// that includes all of [requiredComponents], if there is one.
-  Map<String, dynamic> _visualStudioDetails({Iterable<String> requiredComponents}) {
+  Map<String, dynamic> _visualStudioDetails(
+      {Iterable<String> requiredComponents, List<String> additionalArguments}) {
     final List<String> requirementArguments = requiredComponents == null
         ? <String>[]
         : <String>['-requires', ...requiredComponents];
     try {
-      final ProcessResult whereResult = processManager.runSync(<String>[
-        _vswherePath,
-        '-format', 'json',
+      final List<String> defaultArguments = <String>[
+        '-format',
```

Comment:
Nit: since we're formatting manually anyway, I would prefer keeping `'json'` on the same line with this to make it obvious that it's conceptually one argument.

Reply:
Done. I was wondering why it was indented like that, but makes sense.